src/bidiMapper/domains/context/browsingContextStorage.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 2022 Google LLC.
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BrowsingContextImpl } from './browsingContextImpl';
+import { NoSuchFrameException } from '../protocol/error';
+
+export class BrowsingContextStorage {
+  static #contexts: Map<string, BrowsingContextImpl> = new Map();
+
+  static getTopLevelContexts(): BrowsingContextImpl[] {
+    return Array.from(BrowsingContextStorage.#contexts.values()).filter(
+      (c) => c.parentId === null
+    );
+  }
+
+  static forgetContext(contextId: string) {
```

Comment:
Can we call this `deleteContext`? `forget` is a bit non-standard w.r.t. CRUD terminology.

Reply:
I wanted to underline it removes the context from storage.

Reply:
Renamed `remove` to `delete` and `forget` to `remove`.